components/tidb_query/src/batch/runner.rs

```diff
@@ -389,12 +390,26 @@ impl<SS: 'static> BatchExecutorsRunner<SS> {
                     );
                     // Although `schema()` can be deeply nested, it is ok since we process data in
                     // batch.
-                    result.physical_columns.encode(
-                        &result.logical_rows,
-                        &self.output_offsets,
-                        self.out_most_executor.schema(),
-                        data,
-                    )?;
+                    match self.config.encode_type {
+                        EncodeType::TypeDefault => {
```

Comment:
The `maximum_encoded_size` above is calculated for the binary format instead of the arrow format. You need to write a new `maximum_encoded_size` for the arrow format.

Reply:
Addressed. But I don't have a good idea to calculate the maximum_encoded_size for an undecoded vec. Any idea?

Reply:
I don't either. Maybe considering simply * factor differently for each vector type?